examples/addition_rnn.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from keras.models import Sequential, slice_X
+from keras.layers.core import Activation, Dense, RepeatVector
+from keras.layers import recurrent
+from sklearn.utils import shuffle
+import numpy as np
+
+"""
+An implementation of sequence to sequence learning for performing addition
+Input: "535+61"
+Output: "596"
+Padding is handled by using a repeated sentinel character (space)
+
+By default, the JZS1 recurrent neural network is used
+JZS1 was an "evolved" recurrent neural network performing well on arithmetic benchmark in:
+"An Empirical Exploration of Recurrent Network Architectures"
+http://jmlr.org/proceedings/papers/v37/jozefowicz15.pdf
+
+Input may optionally be inverted, shown to increase performance in many tasks in:
+"Learning to Execute"
+http://arxiv.org/abs/1410.4615
+and
+"Sequence to Sequence Learning with Neural Networks"
+http://papers.nips.cc/paper/5346-sequence-to-sequence-learning-with-neural-networks.pdf
+Theoretically it introduces shorter term dependencies between source and target.
+
+Two digits inverted:
++ One layer JZS1 (128 HN) with 55 iterations = 99% train/test accuracy
+Three digits inverted:
++ One layer JZS1 (128 HN) with 19 iterations = 99% train/test accuracy
+Four digits inverted:
++ One layer JZS1 (128 HN) with 20 iterations = 99% train/test accuracy
+Five digits inverted:
++ One layer JZS1 (128 HN) with 28 iterations = 99% train/test accuracy
```

Comment:
Importantly: in each case, how many *unique* examples were generated? (e.g. 150k in the 3 digits case)